### PR TITLE
dpkg: 1.22.6 -> 1.22.10

### PR DESCRIPTION
**Pull request by siriobalmelli**
## Description of changes

https://github.com/NixOS/nixpkgs/pull/330503 breaks the build on Darwin, this follow-up PR fixes that.

I'm not sure what the correct way would be to propose changes be integrated to an existing PR created by @r-ryantm, so I'm linking both to each other and asking for guidance please 🙏

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -38526,7 +38526,9 @@ with pkgs;
 
   eiciel = callPackage ../tools/filesystems/eiciel { };
 
-  dpkg = callPackage ../tools/package-management/dpkg { };
+  dpkg = callPackage ../tools/package-management/dpkg {
+    inherit (darwin.apple_sdk.frameworks) CoreServices;
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:38531]:
```suggestion
  dpkg = callPackage ../tools/package-management/dpkg { };
```
that is now kinda an anti pattern and we want to stop using it.

File: pkgs/tools/package-management/dpkg/default.nix [hunk 2]
```diff
@@ -71,7 +72,7 @@ stdenv.mkDerivation rec {
        --replace '"ldconfig"' \"${glibc.bin}/bin/ldconfig\"
   '';
 
-  buildInputs = [ perl zlib bzip2 xz zstd libmd ];
+  buildInputs = [ perl zlib bzip2 xz zstd libmd ] ++ lib.optional stdenv.isDarwin [ CoreServices ];
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/dpkg/default.nix:75]:
```suggestion
  buildInputs = [ perl zlib bzip2 xz zstd libmd ]
    ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.CoreServices ];
```

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (siriobalmelli) [reply pkgs/top-level/all-packages.nix]: ok, respun

File: pkgs/tools/package-management/dpkg/default.nix (hunk 2 above)
Comment (siriobalmelli) [reply pkgs/tools/package-management/dpkg/default.nix]: ok, respun

**Comment by siriobalmelli** [comment]
Result of `nixpkgs-review pr 330806` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>sslmate-agent</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>bdf2psf</li>
    <li>cargo-deb</li>
    <li>debian-devscripts</li>
    <li>dpkg</li>
    <li>gohufont</li>
    <li>unifi</li>
    <li>unifi8</li>
    <li>uw-ttyp0</li>
  </ul>
</details>

**Pull request merged**